### PR TITLE
fix issue-8695:DefaultFuture turn off logging optimization for Channel

### LINKED ISSUE #8695: org.apache.dubbo.remoting.exchange.support.DefaultFuture doWithCloseChannel will logRequestData
- [x] I have searched the [issues](https://github.com/apache/dubbo/issues) of this repository and believe that this is not a duplicate.
- [x] I have checked the [FAQ](https://github.com/apache/dubbo/blob/master/FAQ.md) of this repository and believe that this is not a duplicate.

### Environment

* Dubbo version:  2.7.14-SNAPSHOT
* Operating System version: Win10
* Java version: JDK8

### Actual Result
打印出来的日志带有请求toString的内容
```
 public static void closeChannel(Channel channel) {
        for (Map.Entry<Long, Channel> entry : CHANNELS.entrySet()) {
            if (channel.equals(entry.getValue())) {
                DefaultFuture future = getFuture(entry.getKey());
                if (future != null && !future.isDone()) {
                    Response disconnectResponse = new Response(future.getId());
                    disconnectResponse.setStatus(Response.CHANNEL_INACTIVE);
                    disconnectResponse.setErrorMessage("Channel " +
                            channel +
                            " is inactive. Directly return the unFinished request : " +
                            // 这个方法会打印整个 request 对象toString的值
                            future.getRequest());
                    DefaultFuture.received(channel, disconnectResponse);
                }
            }
        }
    }
```
timeout打印的请求对象是不带有Request的
```
    private String getTimeoutMessage(boolean scan) {
        long nowTimestamp = System.currentTimeMillis();
        return (sent > 0 ? "Waiting server-side response timeout" : "Sending request timeout in client-side")
                + (scan ? " by scan timer" : "") + ". start time: "
                + (new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSS").format(new Date(start))) + ", end time: "
                + (new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSS").format(new Date(nowTimestamp))) + ","
                + (sent > 0 ? " client elapsed: " + (sent - start)
                + " ms, server elapsed: " + (nowTimestamp - sent)
                : " elapsed: " + (nowTimestamp - start)) + " ms, timeout: "
                + timeout + " ms, request: " + (logger.isDebugEnabled() ? request : getRequestWithoutData()) + ", channel: " + channel.getLocalAddress()
                + " -> " + channel.getRemoteAddress();
    }
```

这个是漏加了吧

### PR DESCRIPTION
**Pull request by candyaaa**
## What is the purpose of the change

fix #8695 on 3.0

## Brief changelog

DefaultFuture turn off logging optimization for Channel.

### CODE REVIEW

File: dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/exchange/support/DefaultFuture.java [hunk 1]
```diff
@@ -251,14 +251,12 @@ private String getTimeoutMessage(boolean scan) {
...
-                + timeout + " ms, request: " + (logger.isDebugEnabled() ? request : getRequestWithoutData()) + ", channel: " + channel.getLocalAddress()
+                + timeout + " ms, request: " + (logger.isDebugEnabled() ? request : getRequestWithoutData(request)) + ", channel: " + channel.getLocalAddress()
                 + " -> " + channel.getRemoteAddress();
     }
 
-    private Request getRequestWithoutData() {
-        Request newRequest = request.copy();
-        newRequest.setData(null);
-        return newRequest;
+    private static Request getRequestWithoutData(Request request) {
```
Comment (horizonzy) [inline dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/exchange/support/DefaultFuture.java:258]: This method can be removed. We can use request. copyWithoutData() directly.

File: dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/exchange/support/DefaultFuture.java [hunk 2]
```diff
@@ -147,7 +147,7 @@ public static void closeChannel(Channel channel) {
                     disconnectResponse.setErrorMessage("Channel " +
                             channel +
                             " is inactive. Directly return the unFinished request : " +
-                            future.getRequest());
+                            getRequestWithoutData(future.getRequest()));
```
Comment (horizonzy) [inline dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/exchange/support/DefaultFuture.java:150]: Here use `logger.isDebugEnabled()`  to judge is print data also.

**Comment by candyaaa** [comment]
@horizonzy I have completed the optimization you mentioned. This is my first submission, and I want to know about the submission of the issue. How to determine which branch to merge into?

**Comment by horizonzy** [comment]
> @horizonzy I have completed the optimization you mentioned. This is my first submission, and I want to know about the submission of the issue. How to determine which branch to merge into?

Now we use brnach 3.0 as our default branch. And branch master also be maintained, you can pr to master again.

**Review by horizonzy — APPROVED** [review]
LGTM

**Comment by candyaaa** [comment]
> > @horizonzy I have completed the optimization you mentioned. This is my first submission, and I want to know about the submission of the issue. How to determine which branch to merge into?
> 
> Now we use brnach 3.0 as our default branch. And branch master also be maintained, you can pr to master again.

I've submitted to the Master branch again. link https://github.com/apache/dubbo/pull/8778

**Pull request merged**